### PR TITLE
Enable Selenium tests on Travis

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

Enables selenium tests via webdriver to be run Sauce Labs on Travis. 
### How to review

Confirms Travis is running the tests and that they all pass

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,21 +16,22 @@ install:
...
 before_script:
+- npm run compile
 - "./scripts/run_app.sh"
 script:
 - "./scripts/run_tests.sh"
 # fail travis if under X% test coverage
 - coverage report --fail-under=77
 notifications:
--  slack: digitaleq:LDxJXRGeVmg3PpdUhp90eY3Z
+- slack: digitaleq:LDxJXRGeVmg3PpdUhp90eY3Z
```
Comment (benjefferies) [inline .travis.yml:26]: Are the notifications working? travis docs suggests that it should notify every time https://docs.travis-ci.com/user/notifications/ but I haven't got a notification or I am not subscribed to the correct channel?

File: gulp/tests.js [hunk 2]
```diff
@@ -4,26 +4,10 @@ import path from 'path'
...
 import {paths} from './paths'
 import webdriver from 'gulp-webdriver'
-import selenium from 'selenium-standalone'
 import yargs from 'yargs'
 
 const KarmaServer = karma.Server
 
-export let seleniumServer
-
-export function startSeleniumServer(done) {
```
Comment (benjefferies) [inline gulp/tests.js:14]: Is this going to cause issues running the functional tests locally? I am getting ERROR: Couldn't connect to selenium server when running the tests locally.

Comment (benjefferies) [reply gulp/tests.js:14]: Ignore this as I have done a `npm install` and I no longer see this issue

**Comment by iwootten** [comment]
Trying to run this locally all I get is "An element could not be located on the page using the given search parameters." after a npm install. Chrome fires up with "data;" in the address bar in each scenario.

**Comment by ajmaddaford** [comment]
@iwootten you need to run npm install, then run the app (./scripts/run_app.sh) and then run the tests (npm run test_functional).

**Comment by ajmaddaford** [comment]
Looking at the Travis output the selenium tests took 5.27 min, as we increase the number of tests I think this will add too much time to the build. It may be worth thinking about having a subset of selenium tests run on every build and have an overnight job to run through all tests.

File: .travis.yml (hunk 1 above)
Comment (hamishtaplin) [reply .travis.yml:26]: @benjefferies I think we disabled the notifications in Slack itself.

File: gulp/tests.js (hunk 2 above)
Comment (hamishtaplin) [reply gulp/tests.js:14]: @benjefferies Yeah this code was removed as newer versions of webdriver include the Selenium server so you don't have to spin one up manually anymore.

**Comment by hamishtaplin** [comment]
@ajmaddaford What would the subset of tests be?

**Comment by ajmaddaford** [comment]
@hamishtaplin until we get a more complete set of feature tests, possibly just mci and rsi-saverestore-0112?

**Comment by hamishtaplin** [comment]
@ajmaddaford Done.

File: tests/functional/config.js [hunk 3]
```diff
@@ -1,82 +1,57 @@
...
 }
 
 if (process.env.TRAVIS === 'true') {
   config = {
     ...config,
     ...sauceLabsConfig,
-    logLevel: 'debug'
+    logLevel: 'debug',
+    capabilities: [firefox],
+    specs: ['mci', 'rsi-saverestore-0112'].map(spec => `${paths.test.wdioSpec}/${spec}.js`)
```
Comment (iwootten) [inline tests/functional/config.js]: This fails for me:

pattern tests/functional/spec/mci.js did not match any file
pattern tests/functional/spec/rsi-saverestore-0112.js did not match any file

Do you need to specify the ".spec"?

Comment (hamishtaplin) [reply tests/functional/config.js]: @iwootten Oops. Done.

**Review by ajmaddaford — APPROVED** [review]
LGTM 👍

**Pull request merged**